lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -64,27 +65,30 @@ public boolean fileContentsUpdated() {
   }
 
   private synchronized void poll() {
-    long latestChecksumValue = getChecksumValue();
-    if (latestChecksumValue != prevChecksumValue) {
-      prevChecksumValue = latestChecksumValue;
+    Long latestChecksumValue = getChecksumValue();
 
-      // If the latest checksum value is a value other than zero, it means that the file is not
-      // empty.
-      // The if check above ensures that it is also different from the previous checksum, so the
-      // file contents must have been updated.
-      if (latestChecksumValue != 0) {
+    // | latest | previous | outcome                 |
+    // | ------ | -------- | ----------------------- |
+    // | null   | null     | do nothing              |
+    // | null   | !null    | do nothing              |
+    // | !null  | null     | update (no file before) |
+    // | !null  | !null    | update if changed       |
+    if (latestChecksumValue != null) {
+      if (!prevChecksumValue.equals(latestChecksumValue)) {
```

Comment:
`previousCheckSumValue` may be `null` right? So this might thow a `NullPointerException`.
```suggestion
      if (!latestChecksumValue.equals(previousChecksumValue)) {
```

Reply:
great catch! I thought I had done that, but it seems I did not